### PR TITLE
Fixed Hard Fault in ST and TI. Reverted secure element key generation regression.

### PR DESCRIPTION
**Pull request by lundinc2**
Description
-----------

Checklist:
----------

- [ ] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: vendors/st/boards/stm32l475_discovery/aws_demos/config_files/iot_pkcs11_config.h [hunk 1]
```diff
@@ -83,6 +83,15 @@
...
 #define pkcs11configJITP_CODEVERIFY_ROOT_CERT_SUPPORTED    0
 
+/**
+ * @brief Disable ECDSA crypto algorithm (i.e. mechanism) support in this
+ * PKCS #11 module.
+ *
+ * Set this to 1 when the host doesn't support ECDSA for network authentication
+ * (e.g. in the TLS protocol implementation).
+ */
+#define pkcs11configSUPPRESS_ECDSA_MECHANISM               1
```
Comment (dcgaws) [inline vendors/st/boards/stm32l475_discovery/aws_demos/config_files/iot_pkcs11_config.h]: For TI, below, you've got the equivalent change for Test as well as Demo. But for STM32, you've only got it for Demo. Is that intentional?

File: demos/dev_mode_key_provisioning/src/aws_dev_mode_key_provisioning.c [hunk 2]
```diff
@@ -1028,10 +1028,6 @@ CK_RV xProvisionDevice( CK_SESSION_HANDLE xSession,
                 configPRINTF( ( "Warning: could not clean-up old crypto objects. %d \r\n", xResult ) );
             }
         }
-        else
```
Comment (dcgaws) [inline demos/dev_mode_key_provisioning/src/aws_dev_mode_key_provisioning.c]: The change that's being reverted here was an attempt to fix another bug, correct? Do you have an alternative fix for that other bug?

File: vendors/st/boards/stm32l475_discovery/aws_demos/config_files/iot_pkcs11_config.h (hunk 1 above)
Comment (lundinc2) [reply vendors/st/boards/stm32l475_discovery/aws_demos/config_files/iot_pkcs11_config.h]: ST already had this configuration, that is why this issue did not exist on the test project of ST.

File: demos/dev_mode_key_provisioning/src/aws_dev_mode_key_provisioning.c (hunk 2 above)
Comment (lundinc2) [reply demos/dev_mode_key_provisioning/src/aws_dev_mode_key_provisioning.c]: In hindsight this was not the right thing to do, I fixed the bug accidentally, but this code was correct. The real issue was actually the call to generatekey, which tried to generate an ECDSA key on ST and TI, causing a hard fault.

**Pull request merged**